lib/api/core/models/repositories/userRepository.js

```diff
@@ -81,6 +107,31 @@ module.exports = function (kuzzle) {
     return deferred.promise;
   };
 
+  UserRepository.prototype.loadByUsernameAndPassword = function (username, password) {
```

Comment:
Not exactly.
We could also use UserRepository.prototype.load(username), and then compare user's password with the one given by the user. But for this feature especifically, I found cleaner and easier to manage password encryption/decryption within kuzzle kernel, and so just call loadByUsernameAndPassword() from the plugin.

For other authentication plugin, using external services, the strategy will always reponse with a strategy-dependant identifier.
So we will have 2 choices : 
- either map directly this identifier to the kuzzle user's id, and just call UserRepository.prototype.load(id)
- or map it to another kuzzle user's field, and we have to implement only one method UserRepository.prototype.search('identifier-name', id);